### PR TITLE
Current request forwarder encodes query parameters twice.

### PR DESCRIPTION
**Pull request by peteclark-ft**

### CODE REVIEW

File: api-policy-component-service/src/test/java/com/ft/up/apipolicy/ApiPolicyComponentHappyPathsTest.java [hunk 1]
```diff
@@ -750,12 +752,12 @@ public void givenVaryHeaderWithAcceptShouldAddXPolicy() {
...
     private URI sinceSomeDateFromFacade() {
         return fromFacade("/content/notifications")
-                .queryParam("since", "2014-10-15")
+                .queryParam("since", "2017-10-17T15:22:49.804Z")
                 .build();
     }
 
     private void stubForNotifications(String sinceDate, String type, Collection<String> forBrands, Collection<String> notForBrands, String responseBody) throws IOException {
-        MappingBuilder request = get(urlPathEqualTo(NOTIFICATIONS_PATH)).withQueryParam(SINCE, equalTo(sinceDate));
+        MappingBuilder request = get(urlPathEqualTo(NOTIFICATIONS_PATH)).withQueryParam(SINCE, equalTo(URLEncoder.encode(sinceDate, "UTF-8")));
```
Comment (khatton-ft) [inline api-policy-component-service/src/test/java/com/ft/up/apipolicy/ApiPolicyComponentHappyPathsTest.java]: This "looks" wrong, but I think it's necessary if using WireMock 1.x (as we are, for now). I wonder if this asymmetry in encoding is what led to the original change that you've now reverted. See https://github.com/tomakehurst/wiremock/issues/377

But ... you should at least be consistent and use a single `encode` method throughout (see lines 1039, 1069, 1093).

File: api-policy-component-service/src/test/java/com/ft/up/apipolicy/JerseyRequestForwarderTest.java [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+import jersey.repackaged.com.google.common.collect.Sets;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import static org.junit.Assert.*;
+import static org.mockito.Mockito.*;
```
Comment (khatton-ft) [inline api-policy-component-service/src/test/java/com/ft/up/apipolicy/JerseyRequestForwarderTest.java]: Tsk, wildcard imports, and also import ordering (use Google style).

File: api-policy-component-service/src/test/java/com/ft/up/apipolicy/JerseyRequestForwarderTest.java [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+    }
+    
+    @Test
+    public void testForwardConceptSearchRequest() {
+        JerseyRequestForwarder forwarder = new JerseyRequestForwarder(client, endpointConfiguration);
+        
+        MutableRequest request = new MutableRequest(Sets.newHashSet(), "tid_1234");
+        request.setAbsolutePath("/concepts");
+        request.setHttpMethod("GET");
+        request.setRequestEntity("".getBytes());
```
Comment (khatton-ft) [inline api-policy-component-service/src/test/java/com/ft/up/apipolicy/JerseyRequestForwarderTest.java]: seems a convoluted way to write `new byte[0]` (but shouldn't MutableRequest work with a null body anyway?).

File: api-policy-component-service/src/test/java/com/ft/up/apipolicy/JerseyRequestForwarderTest.java [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+        
+        MutableResponse actualResponse = forwarder.forwardRequest(request);
+        assertEquals(HttpServletResponse.SC_OK, actualResponse.getStatus());
+        
+        verify(client).target(conceptSearch);
+        verifyMocks();
+    }
+    
+    @Test
+    public void testForwardNotificationsForConceptRequest() {
```
Comment (khatton-ft) [inline api-policy-component-service/src/test/java/com/ft/up/apipolicy/JerseyRequestForwarderTest.java]: Misleadingly named method?

File: api-policy-component-service/src/test/java/com/ft/up/apipolicy/JerseyRequestForwarderTest.java [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+        queryParams.add("type", "http://www.ft.com/ontology/Topic");
+        queryParams.add("q", "new york");
+        
+        request.setQueryParameters(queryParams);
+        
+        MultivaluedMap headers = new MultivaluedHashMap();
+        headers.add("X-Test-Header", "EXAMPLE");
+        request.setHeaders(headers);
+        
+        URI conceptSearch = URI.create("http://hostname:8080/concepts?mode=search&q=new+york&type=http%3A%2F%2Fwww.ft.com%2Fontology%2Fperson%2FPerson&type=http%3A%2F%2Fwww.ft.com%2Fontology%2Forganisation%2FOrganisation&type=http%3A%2F%2Fwww.ft.com%2Fontology%2FLocation&type=http%3A%2F%2Fwww.ft.com%2Fontology%2FTopic");
```
Comment (khatton-ft) [inline api-policy-component-service/src/test/java/com/ft/up/apipolicy/JerseyRequestForwarderTest.java]: I wasn't convinced that the URLs in query parameter values _need_ to be encoded here, as they aren't in an ambiguous context, but I've delved into the Jersey source code and it seems they will be encoded anyway, so this is correct.

File: api-policy-component-service/src/test/java/com/ft/up/apipolicy/JerseyRequestForwarderTest.java (hunk 3 above)
Comment (peteclark-ft) [reply api-policy-component-service/src/test/java/com/ft/up/apipolicy/JerseyRequestForwarderTest.java]: I'll fix the `MutableRequest` to handle a `null` body, as it currently doesn't

**Pull request merged**